src/burst_db/build_frame_db.py

```diff
@@ -0,0 +1,743 @@
+
+    # do the antimeridian frames first
+    am_idxs = df.geometry.map(_is_on_antimeridian).values
+    epsgs[am_idxs] = df[am_idxs].geometry.map(antimeridian_epsg)
+
+    # everything else
+    # get the x, y (lon, lat) coords of all other rows
+    other_coords = np.array(
+        df[~am_idxs].geometry.map(lambda g: tuple(g.centroid.coords)[0]).tolist()
+    )
+    xs, ys = other_coords.T
+    ys_full_size = np.ones(len(epsgs)) * np.nan
+    ys_full_size[~am_idxs] = ys
+
+    idxs = np.logical_and.reduce((~am_idxs, ys_full_size > 84))
+    epsgs[idxs] = 3413
+
+    idxs = np.logical_and.reduce((~am_idxs, ys_full_size < -60))
+    epsgs[idxs] = 3031
+
+    utm_idxs = np.logical_and(ys < 84, ys > -60)
+    north_idxs = ys[utm_idxs] > 0
+
+    # North hemisphere
+    zones_north = [
+        utm.from_latlon(y, x)[2]
+        for (y, x) in zip(ys[utm_idxs][north_idxs], xs[utm_idxs][north_idxs])
+    ]
+    idxs = np.logical_and.reduce((~am_idxs, ys_full_size < 84, ys_full_size > 0))
+    epsgs[idxs] = 32600 + np.array(zones_north)
+
+    # South hemisphere
+    zones_south = [
+        utm.from_latlon(y, x)[2]
+        for (y, x) in zip(ys[utm_idxs][~north_idxs], xs[utm_idxs][~north_idxs])
+    ]
+    idxs = np.logical_and.reduce((~am_idxs, ys_full_size > -60, ys_full_size < 0))
+    epsgs[idxs] = 32700 + np.array(zones_south)
+
+    # Set all Greenland frames to EPSG:3413
+    geom_greenland = get_greenland_shape()
+    is_in_greenland = get_land_indicator(df, geom_greenland)
+    print(f"{is_in_greenland.sum()} frames are in Greenland. Setting to EPSG:3413")
+    epsgs[is_in_greenland] = 3413
+
+    return epsgs
+
+
+def antimeridian_epsg(mp):
+    """Calculate the EPSG of multipolygons along the antimeridian.
+
+    Parameters
+    ----------
+    mp : shapely.geometry.MultiPolygon
+        The multipolygon to calculate the EPSG for.
+
+    Returns
+    -------
+    epsg : int
+        The EPSG code for the multipolygon.
+
+    Notes
+    -----
+
+    The EPSG code is calculated by taking the weighted average of the centroid of the
+    polygons in the multipolygon. The centroid is weighted by the area of the polygon.
+    The centroid is shifted by 360 degrees if it is in the western hemisphere.
+    """
+    y_c = mp.centroid.y
+    # check north/south pole cases
+    if y_c >= 84.0:
+        return 3413
+    elif y_c <= -60.0:
+        return 3031
+
+    # otherwise, do the weighted average of the shifted polygons to get the centroid
+    A = 0
+    x_weighted = 0
+    # might have 2 or 3 polygons
+    for g in mp.geoms:
+        A += g.area
+        if g.centroid.x < 0:
+            g_shifted = translate(g, xoff=360)
+            x_weighted += g_shifted.centroid.x * g.area
+        else:
+            x_weighted += g.centroid.x * g.area
+    x_c = x_weighted / A
+
+    base = 32600 if y_c > 0 else 32700
+    # longitude 179 gets 32660 north of the equator
+    # longitude -179 gets 32601
+    zone_addition = 1 if x_c < 180 else 60
```

Comment:
great catch! I've updated the comments, since i clearly still confused myself

```python
    # Northern hemisphere = 326XX, southern is 327XX
    base = 32600 if y_c > 0 else 32700
    # EPSG increases negative to positive (west to east)
    # 32601 is at longitude -179 (which 181 after shifting +360)
    zone_addition = 1 if x_c > 180 else 60
    return base + zone_addition
```